cpp/lib/ome/bioformats/FormatReader.h

```diff
@@ -1070,21 +1072,23 @@ namespace ome
        * Get the optimal sub-image width.
        * This is intended for use with openBytes().
        *
+       * @param channel the channel to use.
        * @returns the optimal width.
        **/
       virtual
       dimension_size_type
-      getOptimalTileWidth() const = 0;
+      getOptimalTileWidth(dimension_size_type channel) const = 0;
 
       /**
        * Get the optimal sub-image height.
        * This is intended for use with openBytes().
        *
+       * @param channel the channel to use.
        * @returns the optimal height.
        **/
       virtual
       dimension_size_type
-      getOptimalTileHeight() const = 0;
+      getOptimalTileHeight(dimension_size_type channel) const = 0;
```

Comment:
I agree it's not ideal, it does expose what should really be an implementation detail.  But is this an implementation detail in all circumstances?  If the model is using different sample counts over different channels, and the tiling does alter as a result, then this would be an appropriate way to expose that.

If some channels have different tile/strip sizes due to different sample counts, then removing the argument could lead to suboptimal performance for some channels/planeIndexes, while having the argument means you can pass through the correct sizes in e.g. TiffReader where we can get the values directly from the IFD.  I guess that's one argument in favour of using the dimension number rather than channel--the reader can map the values directly onto the specific IFD for formats which allow that.  Using the highest value of getRGBChannelCount would certainly mitigate the problem though.

I'm not currently decided which way is better.  Both have their points.